src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -53,20 +53,48 @@ private static bool IsEnabled()
                    s_diagnosticListener.IsEnabled();
         }
 
-        private static Activity? CreateActivity(HttpRequestMessage requestMessage)
+        private static Activity? StartActivity(HttpRequestMessage request)
         {
             Activity? activity = null;
             if (s_activitySource.HasListeners())
             {
-                activity = s_activitySource.CreateActivity(DiagnosticsHandlerLoggingStrings.ActivityName, ActivityKind.Client);
+                Uri? requestUri = request.RequestUri;
+                if (requestUri?.IsAbsoluteUri == false)
+                {
+                    requestUri = null;
+                }
+
+                // The following tags might be important for making sampling decisions and should be provided at span creation time.
+                // https://github.com/open-telemetry/semantic-conventions/blob/5077fd5ccf64e3ad0821866cc80d77bb24098ba2/docs/http/http-spans.md?plain=1#L213-L219
+                KeyValuePair<string, object?>[] tags = requestUri is not null
+                    ? new KeyValuePair<string, object?>[4]
+                    : new KeyValuePair<string, object?>[1];
+
+                tags[0] = DiagnosticsHelper.GetMethodTag(request.Method);
+                if (requestUri is not null)
+                {
+                    tags[1] = new("server.address", requestUri.Host);
+                    tags[2] = new("server.port", requestUri.Port);
+                    tags[3] = new("url.full", DiagnosticsHelper.GetRedactedUriString(requestUri));
+                }
+
+                activity = s_activitySource.StartActivity(ActivityKind.Client, name: DiagnosticsHandlerLoggingStrings.ActivityName, tags: tags);
             }
 
-            if (activity is null)
+            if (activity is null &&
+                (Activity.Current is not null ||
+                s_diagnosticListener.IsEnabled(DiagnosticsHandlerLoggingStrings.ActivityName, request)))
             {
-                if (Activity.Current is not null || s_diagnosticListener.IsEnabled(DiagnosticsHandlerLoggingStrings.ActivityName, requestMessage))
+                activity = new Activity(DiagnosticsHandlerLoggingStrings.ActivityName);
+                KeyValuePair<string, object?> methodTag = DiagnosticsHelper.GetMethodTag(request.Method);
```

Comment:
nit: if it's not a known method, the original thing should be recorded in `http.request.method_original`